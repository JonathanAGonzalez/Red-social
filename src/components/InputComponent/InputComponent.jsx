import PropTypes from 'prop-types';
const InputComponent = ({
    labelText,
    id,
    type,
    placeholder,
    onChange,
    value,
}) => {
    return (
        <div className="mb-3">
            <label className="form-label" aria-label={labelText} htmlFor={id} id={id}>
                {labelText}
            </label>
            <input
                className="form-control"
                type={type}
                name={id}
                id={id}
                placeholder={placeholder}
                value={value}
                onChange={(e) => onChange(e.target.value)}
            />
        </div>
    );
};

InputComponent.propTypes = {
    labelText: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
    placeholder: PropTypes.string.isRequired,
};

InputComponent.defaultProps = {
    labelText: '',
    id: '',
    type: 'text',
    placeholder: '',
};

export default InputComponent;
