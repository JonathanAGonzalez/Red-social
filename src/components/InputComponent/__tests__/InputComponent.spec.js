import { fireEvent, render, screen } from '@testing-library/react';
import InputComponent from '../InputComponent';

describe('InputComponent', () => {
  describe('Layout', () => {
    test('should rendered correctly', () => {
      const { asFragment } = render(
        <InputComponent
          labelText="Correo Electronico:"
          id="email"
          type="email"
          placeholder="jona@jona.com"
        />
      );
      expect(asFragment()).toMatchSnapshot();
    });

    test('should have label', () => {
      render(
        <InputComponent
          labelText="Correo Electronico:"
          id="email"
          type="email"
          placeholder="jona@jona.com"
        />
      );
      const label = screen.getByText('Correo Electronico:');
      expect(label).toBeInTheDocument();
    });

    test('should have input', () => {
      render(
        <InputComponent
          labelText="Correo Electronico:"
          id="email"
          type="email"
          placeholder="jona@jona.com"
        />
      );
      const input = screen.getByPlaceholderText('jona@jona.com');
      expect(input).toBeInTheDocument();
    });
  });

  describe('Funcionality', () => {
    test('should acept propertie type', () => {
      render(
        <InputComponent
          labelText="Correo Electronico:"
          id="email"
          type="email"
          placeholder="jona@jona.com"
        />
      );
      const input = screen.getByPlaceholderText('jona@jona.com');
      expect(input.type).toBe('email');
    });

    test('should accept label text', () => {
      render(
        <InputComponent
          labelText="Correo Electronico:"
          id="email"
          type="email"
          placeholder="jona@jona.com"
        />
      );
      const labelText = screen.getByText('Correo Electronico:');
      expect(labelText).toHaveTextContent('Correo Electronico:');
    });

    test('should accept label id', () => {
      render(
        <InputComponent
          labelText="Correo Electronico:"
          id="email"
          type="email"
          placeholder="jona@jona.com"
        />
      );
      const labelText = screen.getByText('Correo Electronico:');
      expect(labelText.id).toBe('email');
    });

    test('should accept input value', () => {
      render(
        <InputComponent
          labelText="Correo Electronico:"
          id="email"
          type="email"
          value="jona@jona.com"
          placeholder="jona@jona.com"
        />
      );
      const input = screen.getByPlaceholderText('jona@jona.com');
      expect(input).toHaveValue('jona@jona.com');
    });

    test('should accept the default value of the property', () => {
      render(
        <InputComponent
          labelText="Correo Electronico:"
          id="email"
          type=""
          value="jona@jona.com"
          placeholder="jona@jona.com"
        />
      );
      const input = screen.getByPlaceholderText('jona@jona.com');
      expect(input.type).toBe('text');
    });

    test('should accept onChange', () => {
      const onChangeMock = jest.fn();
      render(
        <InputComponent
          labelText="Correo Electronico:"
          id="email"
          type="email"
          value="jona@jona.com"
          placeholder="jona@jona.com"
          onChange={onChangeMock}
        />
      );
      const input = screen.getByPlaceholderText('jona@jona.com');

      fireEvent.change(input, {
        target: { value: 'jonathang@hotmail.com.ar' },
      });

      expect(onChangeMock).toHaveBeenCalled();
    });
  });
});
