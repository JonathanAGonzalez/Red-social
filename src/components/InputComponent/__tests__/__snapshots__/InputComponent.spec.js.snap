// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputComponent Layout should rendered correctly 1`] = `
<DocumentFragment>
  <div
    class="mb-3"
  >
    <label
      aria-label="Correo Electronico:"
      class="form-label"
      for="email"
      id="email"
    >
      Correo Electronico:
    </label>
    <input
      class="form-control"
      id="email"
      name="email"
      placeholder="jona@jona.com"
      type="email"
      value=""
    />
  </div>
</DocumentFragment>
`;
